docs/scenarios_authoring/BasicNodes.md

```diff
@@ -98,27 +98,45 @@ There can be at most one edge of type `Default`, and it gets all records that do
 
 ![union_window](img/union_window.png)
 
-Union merges multiple branches into one stream. For each incoming branch two parameters are configured:
-- key - it's value should be of type `String`, definex how elements from branches will be matched together
-- value - this is the output value which will be put the field with name the same as branch id
+Union merges multiple branches into one branch. Events from the incoming branch are passed to the output branch without an attempt to combine them or match them. 
+The #input variable will be no longer available downstream the union node; a new variable will be available instead; it is defined in the union node configuration form.
 
-Union node defines new stream which is union of all branches. In this new stream there is only one variable; it's name is defined by 'Output' parameter; it's value is: 
+
+Branch names visible in the node configuration form are derived from node names preceding the union node.
+
+Example:
+![union_example](img/union_example.png)
+
+Entry fields:
+- Output Variable Name - the name of the variable containing results of the merge.
+- Output Expression - the value of this expression will be passed to the output branch. The output expression is defined separately for each input branch.
+
+Please note, that the #input variable used in the Output expression field refers to the content of the respective incoming branch.
+
+## UnionMemo
+![union_memo_window](img/union_memo_window.png)
+
+Works similarly to Union, but also memoize values for each branches. Memoized state will be cleared when there was no
+subsequent events from any branch during `stateTimeout`.
```

Comment:
There are 2 key differences from the Union (memoization and merge logic) -> I would not say that UnionMemo is similar to Union. 

_Similarly to Union, UnionMemo node merges branches into one branch. There are however important differences in the way UnionMemo works:

- events from the incoming branches are matched together based on some key value
- data which arrived from any of the incoming branch will be memoized by the UnionMemo node for time duration defined in stateTimeout.
- events are emitted after the memoization time (stateTimeout) elapses (????) [on every incoming event ?????\]
- the event time for the emitted events is set to ...........??????
 

-

Reply:
done